go/admin_setbucketaccessmode.go

```diff
@@ -0,0 +1,19 @@
+package bucketclient
+
+import (
+	"context"
+	"fmt"
+)
+
+// AdminSetBucketAccessMode sets the access mode of the given bucket:
+//   - "read-write" to restore the default read/write access
+//   - "read-only" to set the bucket in read-only mode and refuse write
+//     operations with a 503 ServiceUnavailable error.
+//
+// Returns an error if the bucket doesn't exist, or if a request error occurs.
+func (client *BucketClient) AdminSetBucketAccessMode(ctx context.Context,
+	bucketName string, accessMode BucketAccessMode) error {
+	resource := fmt.Sprintf("/_/buckets/%s/accessMode?mode=%s", bucketName, accessMode)
```

Comment:
Should we verify that the `bucketName` does not contain illegal characters (mostly `/`) so as to avoid making an improper call. Not sure this is relevant in this context.

Reply:
It could make sense from a security standpoint indeed, in case a bucket name is crafted voluntarily to call another route on another bucket for example. Since it's an internal route it is probably not an issue but shouldn't hurt to sanitize. Or instead I could URL-encode the bucket name, even though bucket names shouldn't need URL-encoding it's a way to prevent the risk.